src/main/java/qna/domain/Answers.java

```diff
@@ -0,0 +1,49 @@
+package qna.domain;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+import javax.persistence.CascadeType;
+import javax.persistence.Embeddable;
+import javax.persistence.OneToMany;
+
+import qna.CannotDeleteException;
+
+@Embeddable
+public class Answers {
+
+	@OneToMany(mappedBy = "question", cascade = CascadeType.ALL, orphanRemoval = true)
```

Comment:
Question 삭제시 Answers 삭제를 위한거라면

```suggestion
	@OneToMany(mappedBy = "question", cascade = CascadeType.REMOVE, orphanRemoval = true)
```

정도로 작성하면 될 것 같은데, 다른 책임이 있는걸까요?

개인적으로 의도에 맞게 최소한으로만 지정해두는걸 선호합니다 :)

Reply:
첨에 의도햇던건 Question 의 영속상태가 answer 에도 전파되는 것을 의도해서 all 옵션을 걸어두엇는데요.
현 단계의 요구사항만을 고려했을땐 cascade 설정이 필요가 없어서 삭제해두었습니다.
이유는 question 삭제시 question 을 find 해서 영속상태가 되고, 이어서 answers 에 접근에 answer 역시 영속상태가 되는데요.
영속상태에서 상태변경(deleted=true)을 하면 tx 종료시 변경 감지로 인해 db 에 반영되니 이정도만 되어도 요구사항을 충족하게 됩니다.
그래서 굳이 cascade 설정이 현시점에서 필요한가에 대해 의문을 가지면서 삭제하게 되엇습니다.
기능 요구사항은 entity 삭제가 아니라서 remove 설정 역시 제외하게 되었는데요, 혹시 제가 잘못 이해한 부분이 있다면 말씀 부탁드려요 🙏

Reply:
잘 변경해주셨습니다 👍